internal/command/migrate_to_v2/machines.go

```diff
@@ -12,8 +12,16 @@ import (
 	"github.com/superfly/flyctl/internal/machine"
 )
 
-func (m *v2PlatformMigrator) resolveMachineFromAlloc(alloc *api.AllocationStatus) (*api.LaunchMachineInput, error) {
-	mConfig, err := m.appConfig.ToMachineConfig(alloc.TaskName, nil)
+func (m *v2PlatformMigrator) createLaunchMachineInput(oldAllocID string, skipLaunch bool) (*api.LaunchMachineInput, error) {
+	taskName := ""
+
+	if len(m.oldAllocs) > 0 {
+		taskName = m.oldAllocs[0].TaskName
+	} else {
+		taskName = "app"
+	}
```

Comment:
Shouldn't this be looking for the taskName of the alloc specified by `oldAllocID`?
```suggestion
	alloc, hasAlloc := lo.Find(m.oldAllocs, func(alloc *api.AllocationStatus) (*api.AllocationStatus, bool) {
		return alloc, alloc.ID == oldAllocID
	})
	if hasAlloc {
		taskName = alloc.TaskName
	} else {
		taskName = "app"
	}
```

Reply:
yep, you're right!